### ISSUE TITLE
Direct JSON->BSON translation without using CompoundValueNode class

### ISSUE THREAD

**Issue by fgalan** [telefonicaid/fiware-orion#1070]
Currenlty we process compound attribute values in the following way:

JSON -(parsing layer)-> CompoundValueNode -(mongoBackendLayer)-> BSON

This double-processing makes sense in v1 (where input may come eiher in XML or JSON). However, in v2 maybe is more efficient just passing the JSON point to the attribute and make the processing at mongoBackend in just one step removing the CompoundValueNode class, i.e.

JSON -(mongoBackend)-> BSON

This is not going to be implemented soon, just food for thought...

**Effort: 4 man day**

**Comment by kzangeli** [comment]
I guess you mean to try to make the parser just get a string of the compound?
I was looking into that for a few minutes, without success but later I stumbled on something interesting. The method Accept was mentioned. I haven't dedicated any time to this as of right now

**Comment by fgalan** [comment]
Not exactly... What I mean is to apply the same algorithm you are applying now (now = v2, i.e. with rapidjson) at parser layer but in mongoBackend, producing BSON output instead of ComposeValueNode output. The parsing layer would pass the "root rapidjson element" to mongoBackend so mongoBackend could do the processing.

Alterantively, the parser layer could provide BSON to mongoBackend or doing the procesing in third a library if we find that this JSON->BSON process is not suitable neither for parsing layer nor mongoBackend layer.
